cmd/snap-confine/snap-confine.c

```diff
@@ -201,7 +211,7 @@ int main(int argc, char **argv)
 #if 0
 	setup_user_xdg_runtime_dir();
 #endif
-
+	sc_maybe_set_context_environment(snap_context);
```

Comment:
Could you please move this closer to the exec call. Should we ever re-work it to pass environment explicitly that part should be close to change at the same time.